lib/event-subscribers.loader.ts

```diff
@@ -31,39 +45,126 @@ export class EventSubscribersLoader
     const providers = this.discoveryService.getProviders();
     const controllers = this.discoveryService.getControllers();
     [...providers, ...controllers]
-      .filter(wrapper => wrapper.isDependencyTreeStatic())
       .filter(wrapper => wrapper.instance)
       .forEach((wrapper: InstanceWrapper) => {
         const { instance } = wrapper;
         const prototype = Object.getPrototypeOf(instance) || {};
+        const isRequestScoped = !wrapper.isDependencyTreeStatic();
         this.metadataScanner.scanFromPrototype(
           instance,
           prototype,
           (methodKey: string) =>
-            this.subscribeToEventIfListener(instance, methodKey),
+            this.subscribeToEventIfListener(
+              instance,
+              methodKey,
+              isRequestScoped,
+              wrapper.host as Module,
+            ),
         );
       });
   }
 
-  private subscribeToEventIfListener(
+  private async subscribeToEventIfListener(
```

Comment:
Why do we need `async` here? I checked the new added code and it seems we don't use `await` anywhere 🤔

Reply:
Good point, it's not required anymore, I'll fix it!

Reply:
Great catch @delucca!